android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/LoginViewModel.kt

```diff
@@ -87,8 +94,19 @@ class LoginViewModel(
                 when (val result = loginDeferred.awaitWithTimeoutOrNull(LOGIN_TIMEOUT_MILLIS)) {
                     LoginResult.Ok -> {
                         launch {
+                            val isOutOfTimeDeferred = async {
+                                accountRepository.accountExpiryState
+                                    .filterIsInstance<AccountExpiry.Available>()
+                                    .map { it.expiryDateTime.isBeforeNow }
+                                    .first()
+                            }
                             delay(1000)
-                            _uiSideEffect.emit(LoginUiSideEffect.NavigateToConnect)
+                            val outOfTime = isOutOfTimeDeferred.awaitWithTimeoutOrNull(200)
```

Comment:
I don't really understand this, first we wait 1 second and then we wait 200ms more?

Reply:
👍